### PR TITLE
[NATS] Add connection data stream metric types

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Add metric types for connection data stream fields.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #7745

### CODE REVIEW

**Comment by gpop63** [comment]
Mapping

```json
"nats": {
          "properties": {
            "connection": {
              "properties": {
                "idle_time": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "in": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "name": {
                  "type": "keyword",
                  "ignore_above": 1024
                },
                "out": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "pending_bytes": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "subscriptions": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "uptime": {
                  "type": "long",
                  "time_series_metric": "counter"
                }
              }
            },
            "server": {
              "properties": {
                "id": {
                  "type": "keyword",
                  "ignore_above": 1024
                },
                "time": {
                  "type": "date"
                }
              }
            }
          }
        }
```

**Comment by gpop63** [comment]
After fixing aggregation type for counters.

scrnli_9_21_2023_3-01-07 PM

**Comment by gpop63** [comment]
<details>
<summary>Rollover test</summary>

```json
{
  "acknowledged": true,
  "shards_acknowledged": true,
  "old_index": ".ds-metrics-nats.connection-default-2023.09.21-000001",
  "new_index": ".ds-metrics-nats.connection-default-2023.09.21-000002",
  "rolled_over": true,
  "dry_run": false,
  "conditions": {}
}
```

</details> 

<details>
<summary>Mappings</summary>

```json
 "nats": {
          "properties": {
            "connection": {
              "properties": {
                "idle_time": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "in": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "name": {
                  "type": "keyword",
                  "time_series_dimension": true
                },
                "out": {
                  "properties": {
                    "bytes": {
                      "type": "long",
                      "time_series_metric": "counter"
                    },
                    "messages": {
                      "type": "long",
                      "time_series_metric": "counter"
                    }
                  }
                },
                "pending_bytes": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "subscriptions": {
                  "type": "long",
                  "time_series_metric": "gauge"
                },
                "uptime": {
                  "type": "long",
                  "time_series_metric": "counter"
                }
              }
            },
```

</details>

File: packages/nats/data_stream/connection/fields/fields.yml [hunk 1]
```diff
@@ -8,44 +8,52 @@
...
     - name: uptime
       type: long
       format: duration
+      metric_type: counter
       description: |
         The period the connection is up (sec)
     - name: idle_time
       type: long
       format: duration
+      metric_type: gauge
```
Comment (agithomas) [inline packages/nats/data_stream/connection/fields/fields.yml:29]: Can you re-confirm this is Gauge, not counter?

Comment (gpop63) [reply packages/nats/data_stream/connection/fields/fields.yml]: You're right, it's a `counter`.

**Review by agithomas — APPROVED** [review]
LGTM keeping TSDB scope.

**Review by constanca-m — APPROVED** [review]

**Pull request merged**